docs/cli-roadmap.md

```diff
@@ -0,0 +1,82 @@
+1. Multiple objects on command line [#3050](https://github.com/GoogleCloudPlatform/kubernetes/issues/3050)
+1. Finish rolling update [#1353](https://github.com/GoogleCloudPlatform/kubernetes/issues/1353)
+  1. Friendly to auto-scaling [#2863](https://github.com/GoogleCloudPlatform/kubernetes/pull/2863#issuecomment-69701562)
+  1. Rollback
+  1. Rollover
+  1. Write a ReplicationController generator to derive the new ReplicationController from an old one
+  1. Use readiness [#620](https://github.com/GoogleCloudPlatform/kubernetes/issues/620)
+  1. Perhaps factor this in a way that it can be shared with [Openshift’s deployment controller](https://github.com/GoogleCloudPlatform/kubernetes/issues/1743) 
+  1. Rolling update service as a plugin
+1. List supported API versions
+1. List supported resources
+1. Swagger lookups [#3060](https://github.com/GoogleCloudPlatform/kubernetes/issues/3060)
+1. --name, --name-suffix applied during creation and updates
+1. --labels and opinionated label injection: --app=foo, --tier={fe,cache,be,db}, --uservice=redis, --env={dev,test,prod}, --stage={canary,final}, --track={hourly,daily,weekly}, --release=0.4.3c2. Exact ones TBD. We could allow arbitrary values -- the keys are important. The actual label keys would be (optionally?) namespaced with kubectl.kubernetes.io/, or perhaps the user’s namespace.
+1. --annotations and opinionated annotation injection: --description, --revision
+1. Bulk updates (similar to get, create, delete)
+1. Imperative updates. We'll want to optionally make these safe(r) by supporting preconditions based on the current value and resourceVersion.
+  1. label updates: addlabel, rmlabel, changelabel
+  1. annotation updates: addannotation, rmannotation, changeannotation
+  1. more user-friendly on-command-line json for patch
+1. We also want to support the following flavors of more general updates:
+  1. whichever we don’t support:
+    1. safe update: update the full resource, guarded by resourceVersion precondition (and perhaps selected value-based preconditions)
+    1. forced update: update the full resource, blowing away the previous Spec without preconditions; delete and re-create if necessary
+  1. diff/dryrun: Compare new config with current Spec
+  1. submit/apply/reconcile/ensure/merge: Merge user-provided fields with current Spec. Keep track of user-provided fields using an annotation -- see [#1702](https://github.com/GoogleCloudPlatform/kubernetes/issues/1702). Delete all objects with deployment-specific labels.
+1. --dry-run for all commands
+1. Support full label selection syntax, including support for namespaces.
+1. Wait on conditions [#1899](https://github.com/GoogleCloudPlatform/kubernetes/issues/1899)
+1. Make kubectl scriptable: make output and exit code behavior consistent and useful for wrapping in workflows and piping back into kubectl and/or xargs (e.g., dump full URLs?, distinguish permanent and retry-able failure, identify objects that should be retried)
```

Comment:
Here's an example where multiple objects on the command line and an option to dump object names only would be useful in combination: http://techoverflow.net/blog/2013/10/22/docker-remove-all-images-and-containers/